library/list.lisp

```diff
@@ -321,26 +332,33 @@
               (lookup e xs)))))
       ((Nil) None)))
 
-  (declare remove-duplicates (Eq :a => ((List :a) -> (List :a))))
-  (define (remove-duplicates xs)
-    "Returns a new list without duplicate elements."
+  (declare %remove-duplicates (Eq :a => ((List :a) -> (List :a) -> (List :a))))
```

Comment:
i'd prefer a more descriptive name that encodes how this function is different from the exported `remove-duplicates`, like `remove-duplicates-reversing` or something.

Reply:
I've added the `-rev` prefix to the functions that implicitly reverse their first argument. It may not be a bad idea to export something like SRFI 1's `append-reverse` for more efficient list functions.